packages/flutter_tools/lib/src/localizations/localizations_utils.dart

```diff
@@ -149,12 +147,12 @@ class LocaleInfo implements Comparable<LocaleInfo> {
 // See also //master/tools/gen_locale.dart in the engine repo.
 Map<String, List<String>> _parseSection(String section) {
   final Map<String, List<String>> result = <String, List<String>>{};
-  List<String> lastHeading;
+  List<String> lastHeading = <String>[];
   for (final String line in section.split('\n')) {
     if (line == '') {
       continue;
     }
-    if (line.startsWith('  ')) {
+    if (line.startsWith('  ') && lastHeading.isNotEmpty) {
```

Comment:
This was the only thing that wasn't straightforward.

Reply:
I didn't write this, so the logic is pretty new to me as well. Looking at the how it parses the [language registry data](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/localizations/language_subtag_registry.dart), this basically looks for multiline data and appends it to the most recently added heading (`lastHeading`) by assuming that any line that starts with `'  '` is a part of the previous line.

```
Description: Portuguese-Brazilian Orthographic Convention of 1945
  (Convenção Ortográfica Luso-Brasileira de 1945) <-- adds this line to the end of the List<String>
```

I think we could just keep this as it was before?

Reply:
> I didn't write this

Sorry, I was trolled by #65025!  I can pass it off to someone else if you'd prefer (though before that it was last edited in 2018 so you probably still know the most 😄 )

The spirit of this seems to be that `lastHeading` isn't empty since the next line is `lastHeading[lastHeading.length - 1] = ...`.  So if it's malformed or something and there wasn't a previous line it's gonna have a bad time with that insertion.

I can't leave it exactly as it was before and have it be null safe, because it was uninitialized.
The closest to what it was before would be to make it `late`, though again it would have a bad time if something was malformed and `lastHeading` was uninitialized
```dart
  late List<String> lastHeading;
...
    if (line.startsWith('  ')) {
      lastHeading[lastHeading.length - 1] = '${lastHeading.last}${line.substring(1)}';
      continue;
    }
```
Probably better to either make it nullable, then check for null (below) or to make it nonnull as I originally changed it.
```dart
List<String>? lastHeading;
...
    if (line.startsWith('  ') && lastHeading != null) {
      lastHeading[lastHeading.length - 1] = '${lastHeading.last}${line.substring(1)}';
      continue;
    }
```

Reply:
... and just saw your other comments that you prefer `late` here.

Reply:
Yeah, I thought about using late because if the data was malformed, the parsing logic would need to be updated anyway, so might as well fail and have a bad time early on and we can fix it!